src/govuk/components/checkboxes/checkboxes.yaml

```
@@ -77,7 +77,7 @@ params:
   - name: checked
     type: boolean
     required: false
-    description: If `true`, checkbox will be checked.
+    description: If `true`, the checkbox will be checked when the page loads.
```

Is it worth noting that `checked` takes precedent over `values` if set? Same applies to radios and selects. Maybe something like:

```suggestion
    description: If `true`, the checkbox will be checked when the page loads. Setting this option in the checkbox item itself will override any data related to this checkbox item's value in the `values` option, if `values` is set.
```

A bit of a mouthful, I'm sure @EoinShaughnessy can do a better job.

---

What about something like this?

```suggestion
    description: Whether the checkbox should be checked when the page loads. If this option is set the checked state of the checkbox will not be affected by the `value` option.
```

---

Thanks Eoin, that makes sense.

This is one place where I really dislike that we now have a `value` option at the top-level and a `value` option on each item, and they do different things. I think in this context it's not clear which one we're talking about.

I'm wondering if we said 'the top-level `value` option will not affect…' – would that make sense? Is there a better way to differentiate between the two options?

---

Yeah, I see what you mean - would the user know what we meant by 'top-level'?

Sorry for daft question - where in the checkboxes macro options here is the 'top-level' `value` option? The only `value` option I can see is the one with the description, "Value for the checkbox input" on line 62.

Also, are we talking about the `value` or `values` option? The changelog update mentions `values`.

---

Not a daft question at all – what you're asking is pretty much the crux of the issue.

Selects and radios use `value` in two different places to mean two different things:

```
{{ govukRadios({
  name: "colours",
  items: [
    {
      value: "red",
      text: "Red"
    },
    {
      value: "green",
      text: "Green"
    },
    {
      value: "blue", ← [1]
      text: "Blue",
      checked: false ← [3]
    }
  ],
  value: "blue" ← [2]
}) }}
```

(Checkboxes is slightly less ambiguous as it uses `values` at the top level – because multiple checkboxes can be checked when the page first loads so it accepts one or more values.)

We use `item.value` (with a `[1]` in the example) to mean 'the value that should be submitted IF the user chooses this option'.

But we also use `value` at the top-level (`[2]`) to mean 'the value of the checkbox that should be checked when the page loads'.

When we're talking about `value` from the context of the `item.checked` option (`[3]`), I think it's ambiguous whether we're talking about `[1]` or `[2]`. Logically only `[2]` would really makes sense, but it's definitely confusing at the minute.

Hence wondering if 'top-level' is a good way to distinguish between the two…

---

That's a great explanation - thanks! Ok, another pass at rewording:

```suggestion
    description: Whether the checkbox should be checked when the page loads. If you set this option, the `value` option for the checkbox will not affect the checked state.
```